KerbalStuff/blueprints/profile.py

```diff
@@ -12,8 +12,7 @@
 profiles = Blueprint('profile', __name__, template_folder='../../templates/profiles')
 
 FORUM_PROFILE_URL_PATTERN = re.compile(
-    r'^https://forum.kerbalspaceprogram.com/index.php\?/profile/([0-9]+)-(.+)/$')
-
+    r'^(?P<prefix>https?://)?forum.kerbalspaceprogram.com/index.php\?/profile/(?P<id>[0-9]+)-(?P<name>.+)/$')
```

Comment:
Should we make the last `/` at the end optional?
```suggestion
    r'^(?P<prefix>https?://)?forum.kerbalspaceprogram.com/index.php\?/profile/(?P<id>[0-9]+)-(?P<name>.+)/?$')
```

It took me some time, but I did find two profiles where the trailing slash is missing:
- https://spacedock.info/profile/Nils277
- https://spacedock.info/profile/Jatwaa

Reply:
Maybe, but that exact change would add the slash to the name when it's present, so some tweaking would be in order:

```
>>> re.compile(r'^(?P<prefix>https?://)?forum.kerbalspaceprogram.com/index.php\?/profile/(?P<id>[0-9]+)-(?P<name>.+)/?$').match('http://forum.kerbalspaceprogram.com/index.php?/profile/122414-nils277/').groups()
('http://', '122414', 'nils277/')
```

Reply:
Ah I see, so the name capturing group would need to be `(?P<name>[^/]+)` (I'm fairly confident slashes in names aren't allowed, and if they were, they'd have to be escaped for the URL).

Reply:
Yup, and at that point we're getting very little value out of matching the trailing slash or the end of string anyway, might as well just grab all the non-slash characters and be done.